SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h

```diff
@@ -21,20 +21,25 @@ class SPATIALGDK_API USpatialGameInstance : public UGameInstance
 	virtual FGameInstancePIEResult StartPlayInEditorGameInstance(ULocalPlayer* LocalPlayer, const FGameInstancePIEParameters& Params) override;
 #endif
 	virtual void StartGameInstance() override;
+	virtual void Shutdown() override;
 
 	// bResponsibleForSnapshotLoading exists to have persistent knowledge if this worker has authority over the GSM during ServerTravel.
 	bool bResponsibleForSnapshotLoading = false;
 
 	// The SpatialWorkerConnection must always be owned by the SpatialGameInstance and so must be created here to prevent TrimMemory from deleting it during Browse.
 	void CreateNewSpatialWorkerConnection();
 
-	UPROPERTY()
-	USpatialWorkerConnection* SpatialConnection;
-
 protected:
 	// Checks whether the current net driver is a USpatialNetDriver.
 	// Can be used to decide whether to use Unreal networking or SpatialOS networking.
 	bool HasSpatialNetDriver() const;
 	// Helper function that bypasses some of the Unreal flow (which won't work with the SpatialOS model) when launching a new game as a client.
 	bool StartGameInstance_SpatialGDKClient(FString& Error);
+
+private:
+
+	// TODO: Move SpatialConnection ownership to NetDriver
+	friend class USpatialNetDriver;
+	UPROPERTY()
+	USpatialWorkerConnection* SpatialConnection;
```

Comment:
This is going to break server travel in the deployments right? I'm not sure we want to do this at the moment.

Reply:
Yeah if ownership of `SpatialConnection` was moved to the NetDriver that would break `ServerTravel` I think - how to support that would have to be part of a bigger discussion. For now, this shouldn't break anything